mono/utils/mono-threads-state-machine.c

```diff
@@ -742,17 +743,22 @@ Set the no_safepoints flag on an executing GC Unsafe thread.
 The no_safepoints bit prevents polling (hence self-suspending) and transitioning from GC Unsafe to GC Safe.
 Thus the thread will not be (cooperatively) interrupted while the bit is set.
 
-We don't allow nesting no_safepoints regions, so the flag must be initially unset.
+We do not allow nesting no_safepoints regions, so the flag must be initially unset.
 
 Since a suspend initiator may at any time request that a thread should suspend,
 ASYNC_SUSPEND_REQUESTED is allowed to have the no_safepoints bit set, too.
 (Future: We could augment this function to return a return value that tells the
 thread to poll and retry the transition since if we enter here in the
 ASYNC_SUSPEND_REQUESTED state).
+
+This function is internal.
+The public interface is MONO_ENTER_NO_SAFEPOINTS.
  */
 void
-mono_threads_transition_begin_no_safepoints (MonoThreadInfo *info, const char *func)
+mono_threads_enter_no_safepoints_slow (MonoThreadInfo *info, const char *func)
```

Comment:
Don't rename this.  We need to keep a clear distinction between the thread transition functions (the policy). and the layer that combines the policy with the mechanism.
In the case of `mono_threads_enter_no_safepoints_slow`:
1. it should be in `mono-threads-coop.c`
2. it should call `mono_threads_transition_begin_no_safepoints`
3. based on the return value (which is trivial)
4. it should execute the policy (which also trivial and empty in this case).
5. `mono_threads_transition_begin_no_safepoints` should take a non-null `MonoThreadInfo* info` argument - just like the rest of the state machine transition functions.

Let's keep this uniform.


Same for mono_exit_